### PR TITLE
Added route to API for giving feedback rating

### PR DESCRIPTION
**Pull request by samanthaixer**
Created a route for giving feedback on a resource. 
The URL is "/topics/:id/subtopics/:id/resources/:id/feedback"
The post method should be called with a json object containing: {"feedback": 1}. The score could in theory be something other than 1. At the moment this will directly update the database, there is no ability to accumulate a score. However if -1 was sent in this could be used to denote negative feedback. 
Also updated all of the routes to return a header of 500 as an error if the code in the route throws an error.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def topics():
         response.headers.add('Access-Control-Allow-Origin', '*')
         return response
     except Exception as error:
-        return str(error)
+        return str(error), 500
```
Comment (m-budryte) [inline app.py]: Great addition! Are we planning to add more errors (eg 404) in the future?

File: tests/test_controller.py [hunk 2]
```diff
@@ -62,3 +62,42 @@ def test_resources_route():
     data = json.loads(resp.data)
     assert data[0]["name"] == "Array theory"
     assert data[0]["content"] == "Arrays start counting at zero. Trying to access an empty array may throw an out of bounds error"
+
+
+def test_resources_post_feedback():
```
Comment (m-budryte) [inline tests/test_controller.py]: consider extracting and refactoring (possibly thursday?)

Comment (mattTea) [reply tests/test_controller.py]: yaaaaaass!

File: app.py (hunk 1 above)
Comment (mattTea) [reply app.py]: we had 500 in as a catch all, but if you want to handle more error codes in the front we possibly could. 404s already handled if route is not found

**Pull request merged**